core/src/main/java/org/apache/iceberg/FieldMetrics.java

```diff
@@ -78,14 +76,21 @@ public long nanValueCount() {
   /**
    * Returns the lower bound value of this field.
    */
-  public ByteBuffer lowerBound() {
+  public T lowerBound() {
     return lowerBound;
   }
 
   /**
    * Returns the upper bound value of this field.
    */
-  public ByteBuffer upperBound() {
+  public T upperBound() {
     return upperBound;
   }
+
+  /**
+   * Returns if the metrics has bounds (i.e. there is at least non-null value for this field)
+   */
+  public boolean hasBounds() {
+    return upperBound != null;
```

Comment:
I think this should be `valueCount - nullValueCount - nanValueCount > 0`. Otherwise, using `DoubleFieldMetrics.Builder` will always produce bounds, even though it could be only NaN values.

Reply:
Oh, nevermind, I see that `build` handles that case.